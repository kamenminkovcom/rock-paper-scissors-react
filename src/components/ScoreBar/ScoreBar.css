.score-bar-container {
    font-size: 2rem;
    margin-bottom: 1rem;
    list-style: none;
    border-radius: .25rem;
    color: #6c757d;
}