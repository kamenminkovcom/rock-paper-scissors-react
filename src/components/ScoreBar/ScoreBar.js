import React from 'react';
import PropTypes from 'prop-types';

import './ScoreBar.css';

const ScoreBar = ({cpuScores, playerScores}) => (
    <div className="score-bar-container default-row-container">
        <span className="bold">
            CPU: <span>{cpuScores}</span>
        </span>
        <span>Scores</span>
        <span className="bold">
            You: <span>{playerScores}</span>
        </span>
    </div>
);

PropTypes.ScoreBar = {
    cpuScores: PropTypes.number.isRequired,
    playerScores: PropTypes.number.isRequired,
};

export default ScoreBar;