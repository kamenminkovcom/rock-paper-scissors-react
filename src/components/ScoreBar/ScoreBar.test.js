import React from 'react';
import {shallow} from 'enzyme';
import ScoreBar from './ScoreBar';

describe('<ScoreBar/>', () => {
   it('Should render correctly', () => {
        const wrapper = shallow(<ScoreBar cpuScores={1} playerScores={1}/>);
       expect(wrapper).toMatchSnapshot();
   });
});