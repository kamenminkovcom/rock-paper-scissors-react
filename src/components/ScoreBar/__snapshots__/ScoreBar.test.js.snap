// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoreBar/> Should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScoreBar
    cpuScores={1}
    playerScores={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="bold"
>
          CPU: 
          <span>
                    1
          </span>
</span>,
        <span>
          Scores
</span>,
        <span
          className="bold"
>
          You: 
          <span>
                    1
          </span>
</span>,
      ],
      "className": "score-bar-container default-row-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "CPU: ",
            <span>
              1
</span>,
          ],
          "className": "bold",
        },
        "ref": null,
        "rendered": Array [
          "CPU: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "span",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Scores",
        },
        "ref": null,
        "rendered": "Scores",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You: ",
            <span>
              1
</span>,
          ],
          "className": "bold",
        },
        "ref": null,
        "rendered": Array [
          "You: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "span",
          },
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="bold"
>
            CPU: 
            <span>
                        1
            </span>
</span>,
          <span>
            Scores
</span>,
          <span
            className="bold"
>
            You: 
            <span>
                        1
            </span>
</span>,
        ],
        "className": "score-bar-container default-row-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "CPU: ",
              <span>
                1
</span>,
            ],
            "className": "bold",
          },
          "ref": null,
          "rendered": Array [
            "CPU: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "span",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Scores",
          },
          "ref": null,
          "rendered": "Scores",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You: ",
              <span>
                1
</span>,
            ],
            "className": "bold",
          },
          "ref": null,
          "rendered": Array [
            "You: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "span",
            },
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
